source/dataman/HDF5/AMHDF5Group.h

```diff
@@ -0,0 +1,55 @@
+#ifndef AMHDF5GROUP_H
+#define AMHDF5GROUP_H
+
+#include <QObject>
+
+#include <hdf5.h>
+
+#include <QString>
+
+#define AMHDF5GROUP_INVALID_CREATE_OPTION 667000
+#define AMHDF5GROUP_GROUP_ALREADY_OPEN 667001
+#define AMHDF5GROUP_FILE_IN_BAD_STATE 667002
+#define AMHDF5GROUP_GROUP_OPEN_FAILED 667003
+#define AMHDF5GROUP_GROUP_ALREADY_CLOSED 667004
+#define AMHDF5GROUP_GROUP_CLOSE_FAILED 667005
+#define AMHDF5GROUP_GROUP_FLUSH_FAILED 667006
+
+/// This class allows for adding groups to AMHDF5 files. Groups are the highest level of data
+/// organiziation in an HDF5 file and every HDF5 file must have at least one group.
+class AMHDF5Group : public QObject
+{
+	Q_OBJECT
+
+public:
+
+	/// Constructor. Takes the name of the HDF5 Group. Does not create the group itself.
+	explicit AMHDF5Group(const QString &name, QObject *parent = 0);
+	/// Deconstructor for the group object.
+	virtual ~AMHDF5Group();
+	/// Checks if a group is currently open.
+	bool isOpen() const;
+	/// Returns the group ID currently held by AMHDF5Group.
+	hid_t groupID() const { return groupID_;}
```

Comment:
Not a big deal, but I think we could probably make this just `id()` since it is implicitly about an HDF5 group.